File: tests/selenium/pages/login_page.js
const { By, until } = require('selenium-webdriver');

const config = require('../config');
const { BasePage } = require('./base_page');

class LoginPage extends BasePage {
  async openFresh() {
    await this.clearClientState();
    await this.goto('/login');
    await this.waitVisible(By.css('input[type="text"]'));
  }

  async login(username, password) {
    await this.openFresh();
    const usernameInput = await this.waitVisible(By.css('input[type="text"]'));
    const passwordInput = await this.waitVisible(By.css('input[type="password"]'));
    await usernameInput.clear();
    await usernameInput.sendKeys(username);
    await passwordInput.clear();
    await passwordInput.sendKeys(password);
    await this.driver.findElement(By.css('button[type="submit"]')).click();
  }

  async loginExpectDispatcher(username, password) {
    await this.login(username, password);
    await this.driver.wait(
      async () => /\/dispatch\//.test(await this.driver.getCurrentUrl()),
      config.defaultTimeoutMs,
      'Không chuyển vào dashboard dispatcher sau đăng nhập'
    );
  }

  async loginExpectDriver(username, password) {
    await this.login(username, password);
    await this.driver.wait(
      async () => /\/driver\//.test(await this.driver.getCurrentUrl()),
      config.defaultTimeoutMs,
      'Không chuyển vào dashboard tài xế sau đăng nhập'
    );
  }

  async submitBlank() {
    await this.openFresh();
    await this.driver.findElement(By.css('button[type="submit"]')).click();
  }

  async waitStillOnLogin() {
    await this.driver.wait(until.urlContains('/login'), config.defaultTimeoutMs);
  }
}

module.exports = {
  LoginPage
};

File: scripts/qa/TrungChuyen_Autotest/src/test/java/org/example/pages/LoginPage.java
package org.example.pages;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class LoginPage {
    private final WebDriver driver;
    private final WebDriverWait wait;

    private final By usernameInput = By.cssSelector("input[name='username'], input[type='text']");
    private final By passwordInput = By.cssSelector("input[name='password'], input[type='password']");
    private final By loginButton = By.xpath("//button[contains(.,'Đăng nhập') or contains(.,'Dang nhap') or @type='submit']");

    public LoginPage(WebDriver driver, WebDriverWait wait) {
        this.driver = driver;
        this.wait = wait;
    }

    public void loginAsDriver(String username, String password) {
        driver.getCurrentUrl(); // Use driver to avoid unused field warning
        WebElement usernameElement = wait.until(ExpectedConditions.visibilityOfElementLocated(usernameInput));
        usernameElement.clear();
        usernameElement.sendKeys(username);

        WebElement passwordElement = wait.until(ExpectedConditions.visibilityOfElementLocated(passwordInput));
        passwordElement.clear();
        passwordElement.sendKeys(password);

        wait.until(ExpectedConditions.elementToBeClickable(loginButton)).click();
    }
}
